guides/v2.2/coding-standards/technical-guidelines.md

```diff
@@ -715,6 +715,424 @@ You need to read configuration from different sources (like database or filesyst
+
+11.3.7.2. The `before` block SHOULD mostly be `createData` calls with minimal UI interaction.
+
+#### 11.3.8. Test After Block
+
+11.3.8.1. Configuration changes via `magentoCLI` SHOULD occur first in the `after` block.
+
+11.3.8.2. The `after` block SHOULD mostly be `deleteData` calls with minimal UI interaction.
+
+11.3.8.3. **You SHOULD perform the most critical actions first in the `<after>` block.** Perform non-browser driving actions first. These are more likely to succeed as no UI is involved. In the good example, `magentoCLI` and `deleteData` are run first to ensure a proper state. In the bad example, we perform some heavy UI steps first.
+
+**Why?** If something goes wrong there, then the critical `magentoCLI` commands may not get a chance to run, leaving Magento configured incorrectly for any upcoming tests.
+
+{% collapsible Examples: %}
+
+### Recommended
+
+```xml
+<after>
+    <magentoCLI command="indexer:set-mode" arguments="schedule" stepKey="setIndexerMode"/>
+    <magentoCLI command="config:set catalog/frontend/flat_catalog_category 0" stepKey="setFlatCatalogCategory"/>
+    <deleteData createDataKey="category" stepKey="deleteCategory"/>
+    <deleteData createDataKey="createSimpleProduct" stepKey="deleteSimpleProduct"/>
+    <actionGroup ref="AdminDeleteStoreViewActionGroup" stepKey="deleteStoreViewEn">
+        <argument name="customStore" value="customStoreEN"/>
+    </actionGroup>
+    <actionGroup ref="AdminDeleteStoreViewActionGroup" stepKey="deleteStoreViewFr">
+        <argument name="customStore" value="customStoreFR"/>
+    </actionGroup>
+    <actionGroup ref="logout" stepKey="logout"/>
+</after>
+```
+
+### Not recommended
+
+```xml
+<after>
+    <actionGroup ref="AdminDeleteStoreViewActionGroup" stepKey="deleteStoreViewEn">
+        <argument name="customStore" value="customStoreEN"/>
+    </actionGroup>
+    <actionGroup ref="AdminDeleteStoreViewActionGroup" stepKey="deleteStoreViewFr">
+        <argument name="customStore" value="customStoreFR"/>
+    </actionGroup>
+    <deleteData createDataKey="category" stepKey="deleteCategory"/>
+    <deleteData createDataKey="createSimpleProduct" stepKey="deleteSimpleProduct"/>
+    <magentoCLI command="config:set catalog/frontend/flat_catalog_category 0" stepKey="setFlatCatalogCategory"/>
+    <magentoCLI command="indexer:set-mode" arguments="schedule" stepKey="setIndexerMode"/>
+    <actionGroup ref="logout" stepKey="logout"/>
+</after>
+```
+
+{% endcollapsible %}
+
+#### 11.3.9. Test Body
+
+11.3.9.1. The test SHOULD re-use action groups, pages, sections, and elements.
+
+11.3.9.2. Any data created MUST be deleted in the `after` block of the test.
+
+11.3.9.3. `amOnPage` SHOULD be followed by `waitForPageLoad`.
+
+11.3.9.4. `amOnPage` SHOULD use `Page.url` instead of a hardcoded URL.
+
+11.3.9.5. `waitForPageLoad` SHOULD be used instead of `waitForLoadingMaskToDisappear` or `waitForAjaxLoad`.
+
+11.3.9.6. Test file names MUST follow this pattern:
```

Comment:
Same here. Move this file name to 11.3.9.1?

Reply:
Fixed